openscap_daemon/rest_api.py

```diff
@@ -0,0 +1,370 @@
+            return '{"Error": "There are missing fields in the request"}', 400
+        elif content['taskSSG'] == "" or content['taskProfileId'] == "":
+            return '{"Error": "Both taskSSG and taskProfileId fields cannot be empty"}', 400
+        else:
+            task_title = content['taskTitle']
+            task_target = content['taskTarget']
+            task_ssg = content['taskSSG']
+            task_tailoring = content['taskTailoring']
+            task_profile_id = content['taskProfileId']
+            task_online_remediation = content['taskOnlineRemediation']
+            task_schedule_not_before = content['taskScheduleNotBefore']
+            task_schedule_repeat_after = content['taskScheduleRepeatAfter']
+
+            if task_target == "":
+                task_target = "localhost"
+
+            if task_online_remediation not in [1, "y", "Y", "yes"]:
+                task_online_remediation = False
+
+            if task_schedule_not_before == "":
+                task_schedule_not_before_now = datetime.now().strftime("%Y-%m-%d %H:%M")
+                task_schedule_not_before = datetime.strptime(task_schedule_not_before_now,
+                                                             "%Y-%m-%d %H:%M")
+            else:
+                try:
+                    task_schedule_not_before = datetime.strptime(task_schedule_not_before,
+                                                                 "%Y-%m-%d %H:%M")
+                except ValueError:
+                    return '{"Error" : "Invalid taskScheduleNotBefore format. Please use %Y-%m-%d %H:%M format"}'
+
+            if task_schedule_repeat_after == "":
+                task_schedule_repeat_after = 0
+            elif task_schedule_repeat_after == "@daily":
+                task_schedule_repeat_after = 1 * 24
+            elif task_schedule_repeat_after == "@weekly":
+                task_schedule_repeat_after = 7 * 24
+            elif task_schedule_repeat_after == "@monthly":
+                task_schedule_repeat_after = 30 * 24
+            else:
+                task_schedule_repeat_after = 0
+
+            task_id = self.system.create_task()
+            self.system.set_task_title(task_id, str(task_title))
+            self.system.set_task_target(task_id, task_target)
+            self.system.set_task_input(task_id, task_ssg if task_ssg != "" else None)
+            self.system.set_task_tailoring(task_id, task_tailoring if task_tailoring != "" else None)
+            self.system.set_task_profile_id(task_id, task_profile_id)
+            self.system.set_task_online_remediation(task_id, task_online_remediation)
+            self.system.set_task_schedule_not_before(task_id, task_schedule_not_before)
+            self.system.set_task_schedule_repeat_after(task_id, task_schedule_repeat_after)
+            task = [{'id': task_id, 'enabled': '0'}]
+            create_json = '{"tasks":' + json.dumps(task, indent=4) + '}'
```

Comment:
I suggest to refactor this becuase this code is copied from the dbus API part. We should reuse a function here.

Reply:
Do you mean a single function handling these two operations?

Reply:
@mpreisler Hey, could you provide guidance for this specific task?

Thanks

Reply:
Not one function handling both APIs, rather some helper functions used in both REST and dbus parts of the codebase.

For example the code that converts `@weekly` to 7 * 24 hours should be a function used in both places.

Sorry for not being clearer before.